lib/ddtrace/span.rb

```diff
@@ -101,11 +115,8 @@ def set_tag(key, value = nil)
       # Keys must be unique between tags and metrics
       @metrics.delete(key)
 
-      # Ensure `http.status_code` is always a string so it is added to
-      #   @meta instead of @metrics
-      # DEV: This is necessary because the agent looks to `meta['http.status_code']` for
-      #   tagging necessary metrics
-      value = value.to_s if key == Ext::HTTP::STATUS_CODE
+      # DEV: This is necessary because the agent looks at `meta[key]`, not `metrics[key]`.
+      value = value.to_s if ENSURE_AGENT_TAGS.key?(key)
```

Comment:
any reason you can't just do `ENSURE_AGENT_TAGS[key]`?

Reply:
Not really, and I tried your suggestion and it's faster, so I'm changing it to `Hash#[]` 👍